src/widget/wtrackmenu.cpp

```diff
@@ -2170,6 +2210,28 @@ void WTrackMenu::slotClearHotCues() {
             &trackOperator);
 }
 
+void WTrackMenu::slotSortHotcuesByPosition(HotcueSortMode sortMode) {
+    QString progressLabelText;
+    switch (sortMode) {
+    case HotcueSortMode::RemoveOffsets:
+        progressLabelText = tr(
+                "Sorting hotcues of %n track(s) by position (remove offsets)",
+                "",
+                getTrackCount());
+        break;
+    case HotcueSortMode::KeepOffsets:
+    default:
+        progressLabelText =
+                tr("Sorting hotcues of %n track(s) by position", "", getTrackCount());
+        break;
+    }
```

Comment:
lets avoid the `default`. That we get warnings in case you add another sort mode and don't update the switch-case here.
```suggestion
    case HotcueSortMode::KeepOffsets:
        progressLabelText =
                tr("Sorting hotcues of %n track(s) by position", "", getTrackCount());
        break;
    }
```

Reply:
I guess you mean to separate the `default` branch and print a warning, or even add a DEBUG_ASSERT there?

Reply:
No, just remove it completely. See this [godbolt](https://compiler-explorer.com/#g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,selection:(endColumn:1,endLineNumber:29,positionColumn:1,positionLineNumber:29,selectionStartColumn:1,selectionStartLineNumber:29,startColumn:1,startLineNumber:29),source:'enum+class+E+%7B%0A++++A,%0A++++B,%0A%7D%3B%0A%0Aint+foo(E+v)+%7B%0A++++switch+(v)+%7B%0A++++//+commenting+out+one+of+the+below+cases+will+result+in+a+compile+error+(as+intended)%0A++++case+E::A:%0A++++++++return+42%3B%0A++++case+E::B:%0A++++++++return+43%3B%0A++++%7D%0A++++//+the+return+afterwards+is+needed+because+the+technical+definition+of+enums+is+weird+(to+allow+for+bitflags)+%0A++++return+0%3B%0A%7D%0A%0A%0Aint+bar(E+v)+%7B%0A++++switch+(v)+%7B%0A++++//+while+the+same+doesn!'t+happen+when+a+%60default%60+case+is+present:%0A++++//+case+E::A:%0A++++//+++++return+42%3B%0A++++default:%0A++++case+E::B:%0A++++++++return+43%3B%0A++++%7D%0A%7D%3B%0A'),l:'5',n:'0',o:'C%2B%2B+source+%231',t:'0')),k:50,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((g:!((h:compiler,i:(compiler:g142,filters:(b:'0',binary:'1',binaryObject:'1',commentOnly:'0',debugCalls:'1',demangle:'0',directives:'0',execute:'1',intel:'0',libraryCode:'0',trim:'1',verboseDemangling:'0'),flagsViewOpen:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,libs:!(),options:'-Wall+-Werror',overrides:!(),selection:(endColumn:1,endLineNumber:1,positionColumn:1,positionLineNumber:1,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:1),l:'5',n:'0',o:'+x86-64+gcc+14.2+(Editor+%231)',t:'0')),k:50,l:'4',m:50,n:'0',o:'',s:0,t:'0'),(g:!((h:output,i:(compilerName:'x86-64+gcc+14.2',editorid:1,fontScale:14,fontUsePx:'0',j:1,wrap:'1'),l:'5',n:'0',o:'Output+of+x86-64+gcc+14.2+(Compiler+%231)',t:'0')),header:(),l:'4',m:50,n:'0',o:'',s:0,t:'0')),k:50,l:'3',n:'0',o:'',t:'0')),l:'2',n:'0',o:'',t:'0')),version:4) for a demonstration.

Reply:
Okay, so you say the compiler warning is sufficient/better (as long as they're fatal)

Reply:
yes, much better than no warning and a best guess runtime handling. The more we can do/know at compile-time the better.